src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -195,63 +195,75 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
 
 void mp::LXDVMImageVault::prune_expired_images()
 {
-    auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images?recursion=1").arg(base_url.toString())));
-
-    auto images = json_reply["metadata"].toArray();
-
-    for (const auto image : images)
+    try
     {
-        auto image_info = image.toObject();
-        auto last_used = std::chrono::system_clock::time_point(std::chrono::milliseconds(
-            QDateTime::fromString(image_info["last_used_at"].toString(), Qt::ISODateWithMs).toMSecsSinceEpoch()));
+        auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images?recursion=1").arg(base_url.toString())));
+        auto images = json_reply["metadata"].toArray();
 
-        if (last_used + days_to_expire <= std::chrono::system_clock::now())
+        for (const auto image : images)
         {
-            mpl::log(mpl::Level::info, category,
-                     fmt::format("Source image \'{}\' is expired. Removing it…",
-                                 image_info["properties"].toObject()["release"].toString()));
+            auto image_info = image.toObject();
+            auto last_used = std::chrono::system_clock::time_point(std::chrono::milliseconds(
+                QDateTime::fromString(image_info["last_used_at"].toString(), Qt::ISODateWithMs).toMSecsSinceEpoch()));
 
-            lxd_request(
-                manager, "DELETE",
-                QUrl(QString("%1/images/%2").arg(base_url.toString()).arg(image_info["fingerprint"].toString())));
+            if (last_used + days_to_expire <= std::chrono::system_clock::now())
+            {
+                mpl::log(mpl::Level::info, category,
+                         fmt::format("Source image \'{}\' is expired. Removing it…",
+                                     image_info["properties"].toObject()["release"].toString()));
+
+                lxd_request(
+                    manager, "DELETE",
+                    QUrl(QString("%1/images/%2").arg(base_url.toString()).arg(image_info["fingerprint"].toString())));
+            }
         }
     }
+    catch (const LXDNotFoundException&)
+    {
+        return;
+    }
```

Comment:
This feels a bit broad… If the `DELETE` request returns 404, should we still just ignore it?

Reply:
Right, I was too focused on the first `lxd_request()`.  Will work on a better solution.

Reply:
Ok, this should be better now.